Antidote/CallsManager.m

```diff
@@ -451,13 +465,33 @@ - (void)switchViewControllerForCall:(OCTCall *)call
 
     self.currentCallViewController = abstractVC;
 
-    self.currentCall = call;
+    self.currentCall = [RBQSafeRealmObject safeObjectFromObject:call];
 }
 
-- (BOOL)onlyPauseCallsLeft
+- (void)replaceCurrentCallThatWasRemoved
 {
-    return (([self.allActiveCallsController.fetchedObjects count] == 0) &&
-            ([self.allPausedCallsController.fetchedObjects count] != 0));
+    OCTCall *call = [self.allPausedCallsController.fetchedObjects firstObject];
+
+    if (! call) {
+        call = [self.allCallsController.fetchedObjects firstObject];
+    }
+
+    // workaround for deadlock in objcTox
+    // https://github.com/Antidote-for-Tox/objcTox/issues/51
+    [self performSelector:@selector(switchViewControllerForCall:) withObject:call afterDelay:0];
+}
+
+- (BOOL)isCurrentViewControllerReusableForCurrentCall:(OCTCall *)call
+{
+    if (! ((call.status == OCTCallStatusActive) || (call.status == OCTCallStatusPaused))) {
```

Comment:
Call A is active in progress. 
Call B is incoming.

User decides to end Call A, while ignoring Call B's incoming notification.

This is when it is necessary for us to check if the left over calls are, in which case is Call B, can be used with the current view controller.